docs/docs/testing-components-with-graphql.md

```diff
@@ -0,0 +1,321 @@
+              "date": "28 May, 2015",
+              "title": "New Beginnings"
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+GraphiQL doesn't know about any fragments defined by Gatsby, so if your query
+uses them then you'll need to replace those with the content of the fragment. If
+you're using `gatsby-transformer-sharp` you'll find the fragments in
+[gatsby-transformer-sharp/src/fragments.js](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-sharp/src/fragments.js).
+So, for example if your query includes:
+
+```graphql
+    image {
+        childImageSharp {
+            fluid(maxWidth: 1024) {
+                ...GatsbyImageSharpFluid
+            }
+        }
+    }
+```
+
+...it becomes:
+
+```graphql
+    image {
+        childImageSharp {
+            fluid(maxWidth: 1024) {
+                base64
+                aspectRatio
+                src
+                srcSet
+                sizes
+            }
+        }
+    }
+```
+
+When you have the result, copy the `data` value from the output panel. Good
+practice is to store your fixtures in a separate file, but for simplicity here
+you will be defining it directly inside your test file:
+
+```js
+// src/__tests__/index.js
+
+import React from "react"
+import renderer from "react-test-renderer"
+import BlogIndex from "../pages/index"
+
+describe("BlogIndex", () =>
+  it("renders correctly", () => {
+    const location = {
+      pathname: "",
+    }
+
+    const data = {
+      site: {
+        siteMetadata: {
+          title: "Gatsby Starter Blog",
+        },
+      },
+      allMarkdownRemark: {
+        edges: [
+          {
+            node: {
+              excerpt:
+                "Far far away, behind the word mountains, far from the countries Vokalia and\nConsonantia, there live the blind texts. Separated they live in…",
+              fields: {
+                slug: "/hi-folks/",
+              },
+              frontmatter: {
+                date: "28 May, 2015",
+                title: "New Beginnings",
+              },
+            },
+          },
+        ],
+      },
+    }
+
+    const tree = renderer
+      .create(<BlogIndex location={location} data={data} />)
+      .toJSON()
+    expect(tree).toMatchSnapshot()
+  }))
+```
+
+Run the tests and they should now pass. Take a look in `__snapshots__` to see
+the output.
+
+## Testing StaticQuery
+
+The method above works for page queries, as you can pass the data in directly to
+the component. This doesn't work for components that use `StaticQuery` though,
+so we need to take a slightly different approach to testing these. The blog
+starter project doesn't include `StaticQuery`, so the example here is from
+[the StaticQuery docs](/docs/static-query/).
+
+The pattern for enabling type checking described in the docs is a good starting
+point for making these components testable, but you need access to the
+component.
+
+Here is the example:
+
+```js
```

Comment:
What kind of file should someone use this code example for? Maybe readers will just know, but wanted to make sure if it's supposed to go in a specific type of file or you have a suggestion, make it clear here as you did for other code examples!

Reply:
As in where they'd be using StaticQuery rather than a page query? No problem.